File: JavaScript/src/Internals/Functions/DateTimes/Function_ADDHOURS.js
import { Function_2 } from '../Function_2.js';
import { MyDate } from '../../MyDate.js';

import { Operand } from '../../../Operand.js';

class Function_ADDHOURS extends Function_2 {
    get Name() {
        return "AddHours";
    }

    constructor(z) {
    super(z);
  }

    evaluate(engine, tempParameter) {
        let args1 = this.getDate_1(engine, tempParameter);
        if (args1.IsError) { return args1; }

        let args2 = this.getNumber_2(engine, tempParameter);
        if (args2.IsError) { return args2; }
        let date = new Date(args1.DateValue.ToDateTime().getTime());
        date.setHours(date.getHours() + args2.IntValue);
        return Operand.Create(new MyDate(date));
    }
}

export { Function_ADDHOURS };

File: csharp/ToolGood.Algorithm/Internals/Functions/DateTimes/Function_ADDHOURS.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.DateTimes
{
    internal class Function_ADDHOURS : Function_2
    {
		public Function_ADDHOURS(FunctionBase[] funcs) : base(funcs)
		{
		}

		

        public override string Name => "AddHours";

        public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = GetDate_1(engine, tempParameter);
            if (args1.IsError) { return args1; }

            var args2 = GetNumber_2(engine, tempParameter);
            if (args2.IsError) { return args2; }
            var date = args1.DateValue.AddHours(args2.IntValue);
            return Operand.Create(date);
        }

    }

}
